lib/application.js

```diff
@@ -241,6 +244,66 @@ app.use = function use(fn) {
 };
 
 /**
+ * Proxy `Router#disuse()` to remove middleware from the app router.
+ * See Router#disuse() documentation for details.
+ *
+ * If the _fn_ parameter is an express app, then it will be
+ * dismounted from the _route_ specified.
+ *
+ * @public
+ */
+
+app.disuse = function use(fn) {
+  var offset = 0;
+  var path = '/';
+
+  // default path to '/'
+  // disambiguate app.use([fn])
+  if (typeof fn !== 'function')  {
+    var arg = fn;
+
+    while (Array.isArray(arg) && arg.length !== 0)  {
+      arg = arg[0];
+    }
+
+    // first arg is the path
+    if (typeof arg !== 'function')  {
+      offset = 1;
+      path = fn;
+    }
+  }
+
+  var fns = flatten(slice.call(arguments, offset));
+
+  if (fns.length === 0)  {
+    throw new TypeError('app.use() requires middleware functions');
+  }
+
+  // setup router
+  this.lazyrouter();
+  var router = this._router;
+
+  fns.forEach(function (fn)  {
+    // non-express app
+    if (!fn || !fn.handle || !fn.set)  {
+      return router.disuse(path, fn);
+    }
+
+    debug('.disuse app under %s', path);
```

Comment:
> but I do think that the stack would need to be cleaned at a later stage to stop the stack size from growing too large as a possibilitiy

Correct. This would, of course, be a requirement for this to even make it into the code. Let me know when you can implement that :)

> as afar as I can tell during the app mounting using app.use(anotherApp), the only things that seem to be set up are the parent and mountpath properties, which I do remove with disuse.

You may not be too familiar, it seems, and that's OK! That's why we are here to review the code :) The code you need to implement the opposite of is here: https://github.com/strongloop/express/blob/4.13.0/lib/application.js#L89-L100